src/OpenTelemetry.AutoInstrumentation/Configuration/EnvironmentConfigurationTracerHelper.cs

```diff
@@ -97,8 +97,18 @@ public static TracerProviderBuilder AddWcfInstrumentation(TracerProviderBuilder
         }
 
         [MethodImpl(MethodImplOptions.NoInlining)]
-        public static TracerProviderBuilder AddHttpClientInstrumentation(TracerProviderBuilder builder, PluginManager pluginManager)
+        public static TracerProviderBuilder AddHttpClientInstrumentation(TracerProviderBuilder builder, PluginManager pluginManager, LazyInstrumentationLoader lazyInstrumentationLoader)
         {
+            new HttpClientInitializer(lazyInstrumentationLoader, pluginManager);
```

Comment:
This looks awkward when compared to other `Add*Instrumentation` methods. The `HttpClientInitializer` also looks different. Can you refactor this so that the code looks more similar to others?

Reply:
Agree, I'll try to rethink it again. Hoped maybe someone had better vision.

Reply:
The alternative idea I had was to change `InstrumentationInitializer.RequiredAssemblyName` from `string` to `string[]`, but it wasn't anymore intuitive, if `RequiredAssemblyNames` would be any or all required. Wrapping with custom class leaves more flexibility, what ever is necessary for the instrumentation (means its not limiting).

Reply:
I think I initially proposed to separate/decouple the "initialization" from "library detection" (I almost never like `abstract class` 😉 ). In the past we wanted to encapsulate everything in one class (it made sense back then). Now we see that it is hard (at least with current design). Probably we should refactor the design... How about changing `InstrumentationInitializer` to:

```csharp
interface IInstrumentationInitiazer
{
   bool Detect(string loadedAssembly); // returns when the required assemblies are loaded
   void Initialize(ILifespanManager lifespanManager); 
}
```

Reply:
Detection is async pattern, not sure it's good to synchronize it (return bool).

Reply:
Aren't you synchronizing it now anyway? 😉 

You can always make a "lock-free synchronization" when implementing an `IInstrumentationInitiazer`

Reply:
The `new ...` at the beginning is kind of awkward, but, refactoring the code will require some thought and discussion. Let's pursue it in a separate PR/issue.

Reply:
> Aren't you synchronizing it now anyway? 😉
> 
> You can always make a "lock-free synchronization" (when you need multiple assemblies) when implementing an `IInstrumentationInitiazer`

Yes, but the critical difference is where it's synchronized. I'm basically synchronizing the initialize part but synchronizing detect has some effect on the detection itself - you can miss the event and then must choose more expensive path.

Reply:
Issue created https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/1638